solutions/devpsrint-michael-douglas-1/GitHubApp/Screens/Components/Buttons/ButtonView/ButtonView.swift

```diff
@@ -0,0 +1,79 @@
+//
+//  ButtonView.swift
+//  GitHubApp
+//
+//  Created by Gustavo Rocha on 24/01/22.
+//
+
+import UIKit
+
+@IBDesignable
+class ButtonView: UIView {
+    
+    // MARK: Properties
+    private lazy var button: UIButton = {
+        let button = UIButton()
+        button.translatesAutoresizingMaskIntoConstraints = false
+        button.titleLabel?.font =  UIFont.systemFont(ofSize: 16, weight: .semibold)
+        return button
+    }()
+    
+    @IBInspectable
+    var titleColor: UIColor = .orange {
+        didSet {
+            updateUI()
+        }
+    }
+    
+    @IBInspectable
+    var primaryColor: UIColor = .orange {
+        didSet {
+            updateUI()
+        }
+    }
+    
+    @IBInspectable
+    var cornerRadius: Double = 0.0 {
+        didSet {
+            updateUI()
+        }
+    }
+    
+    // MARK: Initialization
+    init() {
+        super.init(frame: .zero)
+        setupUI()
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    // MARK: Methods
+    private func setupUI() {
+        backgroundColor = .white
+        addSubviews()
+        addConstraints()
+    }
+    
+    private func addSubviews() {
+        addSubview(button)
+    }
+    
+    private func addConstraints() {
+        button.topAnchor.constraint(equalTo: topAnchor).isActive = true
+        button.leftAnchor.constraint(equalTo: readableContentGuide.leftAnchor).isActive = true
+        button.rightAnchor.constraint(equalTo: readableContentGuide.rightAnchor).isActive = true
+        button.bottomAnchor.constraint(equalTo: bottomAnchor).isActive = true
```

Comment:
Que tal utilizar desta forma aqui?

```suggestion
        NSLayoutConstraint.activate([
            button.topAnchor.constraint(equalTo: topAnchor),
            button.leftAnchor.constraint(equalTo: readableContentGuide.leftAnchor),
            button.rightAnchor.constraint(equalTo: readableContentGuide.rightAnchor),
            button.bottomAnchor.constraint(equalTo: bottomAnchor)
        ])
```

Reply:
Implementei do modo acima apenas por costume, mas gostaria de saber qual seria a vantagem/desvantagem em comparação a segunda forma que você recomendou ?

Reply:
A segunda forma tem a vantagem de que ela já ativa todas as constraints que estão dentro do `activate([])`. então não corre o risco de tu acabar esquecendo de dar um `isActivate`. Não existe um certo ou errado, é uma forma mais prática 🚀